### PR TITLE
[MM-48499] Custom Emoji delete modal does not use theme properly

### PR DESCRIPTION
**Pull request by kostaspt**
#### Summary
Similar to https://github.com/mattermost/mattermost-webapp/pull/11867 [[MM-40111](https://mattermost.atlassian.net/browse/MM-40111)]. Maybe we should refactor this at some point to make it more generic?

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-48499

#### Related Pull Requests
N/A

#### Screenshots
|  before  |  after  |
|----|----|
| image | image |

#### Release Note
```release-note
NONE
```

[MM-40111]: https://mattermost.atlassian.net/browse/MM-40111?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by BenCookie95 — APPROVED** [review]

**Review by mkraft — APPROVED** [review]

File: components/emoji/emoji_page.tsx [hunk 1]
```diff
@@ -33,6 +35,11 @@ export default class EmojiPage extends React.PureComponent<Props> {
     componentDidMount() {
         this.updateTitle();
         this.props.actions.loadRolesIfNeeded(['system_admin', 'team_admin', 'system_user', 'team_user']);
+        Utils.resetTheme();
+    }
+
+    componentWillUnmount() {
+        Utils.applyTheme(this.props.currentTheme);
```
Comment (M-ZubairAhmed) [inline components/emoji/emoji_page.tsx:42]: Question : how were you able to find out that this is the problem. and why is the theme not being applied in the component in the first place 🤔

Comment (M-ZubairAhmed) [reply components/emoji/emoji_page.tsx:42]: Oh I think this is a good approach

**Review by furqanmlk — COMMENTED** [review]
Hi @kostaspt 3 test cases are failing [Test dashboard](https://automation-dashboard.vercel.app/cycle/290853). Can you fix these?

**Review by coltoneshaw — APPROVED** [review]
Looks good to me!

**Pull request merged**